src/cli.rs

```diff
@@ -340,6 +348,17 @@ pub struct Inspect {
     /// Output the snapshot raw bytes directly, with no base64 encoding
     #[clap(long, requires("dump-snapshot-to"))]
     pub dump_raw_snapshot: bool,
+    /// Query the authorizer after evaluation. If no authorizer is provided, query the token after evaluation.
+    #[clap(
+        long,
+        value_parser = clap::builder::ValueParser::new(parse_rule),
+        requires("public-key"),
+        requires("public-key-file"),
+    )]
+    pub query: Option<Rule>,
+    /// Query facts from all blocks (not just authority, authorizer or explicitly trusted blocks). Be careful, this can return untrustworthy facts.
+    #[clap(long, requires("query"))]
+    pub query_all: bool,
```

Comment:
why is "query" optional but "query_all" is not? It seem it depends on "query" via "requires"...